commit cb83542b039ff7a1b9fba6256ac1eb28afb044ff (refs/changes/71/1671/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-03-16T17:51:31+01:00 (2 years, 7 months ago)
    
    TOOLS-1724 sdcadm trying to read image_uuid of undefined
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 2cca542..46e87c8 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -2785,7 +2785,7 @@ SdcAdm.prototype.getCandidateImages = function getCandidateImages(opts, cb) {
             currImgUuids = Object.keys(currImgUuids);
             if (currImgUuids.length === 0) {
                 // No insts -> use the image_uuid set on the service.
-                if (opts.service.params.image_uuid) {
+                if (opts.service.params && opts.service.params.image_uuid) {
                     currImgUuids.push(opts.service.params.image_uuid);
                 }
             }
@@ -3512,7 +3512,8 @@ SdcAdm.prototype.genUpdatePlan = function genUpdatePlan(options, cb) {
                                 ch.service.name !== 'sdcadm') {
                             // No insts -> use the image_uuid set on the
                             // service.
-                            if (ch.service.params.image_uuid) {
+                            if (ch.service.params &&
+                                ch.service.params.image_uuid) {
                                 currImgUuids.push(ch.service.params.image_uuid);
                             }
                         }
